src/utils.ts

```diff
@@ -223,8 +229,87 @@ export function computeStatus(
         .subtract(medicationRequestExpirationPeriodInDays, "day")
     )
   ) {
-    return "expired";
+    return MedicationRequestStatus.expired;
+  }
+
+  return MedicationRequestStatus.active;
+}
+
+/**
+ * Fetch the "recorded" extension off a medication request
+ * @param medicationDispense
+ */
+export function getDateRecorded(medicationDispense: MedicationDispense) {
+  return medicationDispense?.extension?.find(
+    (ext) => ext.url === OPENMRS_FHIR_EXT_RECORDED
+  )?.valueDateTime;
+}
+
+/**
+ * Given a medication request and an array of medication dispenses, fetch all dispenses authorized by that request
+ *
+ * @param medicationRequest
+ * @param medicationDispenses
+ */
+export function getAssociatedMedicationDispenses(
+  medicationRequest: MedicationRequest,
+  medicationDispenses: Array<MedicationDispense>
+) {
+  return medicationDispenses?.filter((medicationDispense) =>
+    medicationDispense?.authorizingPrescription?.some((prescription) =>
+      prescription.reference.endsWith(medicationRequest.id)
+    )
+  );
+}
+
+/**
+ * Given a set of medication requests, return the status of the one with the most recent recorded date
+ */
+export function getMostRecentMedicationDispenseStatus(
+  medicationDispenses: Array<MedicationDispense>
+) {
+  const sorted = medicationDispenses?.sort(
+    sortMedicationDispensesByDateRecorded
+  );
+  return sorted && sorted.length > 0 ? sorted[0].status : null;
+}
+
+export function sortMedicationDispensesByDateRecorded(
+  a: MedicationDispense,
+  b: MedicationDispense
+) {
+  const dateDiff =
+    parseDate(getDateRecorded(b)).getTime() -
+    parseDate(getDateRecorded(a)).getTime();
+  if (dateDiff !== 0) {
+    return dateDiff;
+  } else {
+    return a.id.localeCompare(b.id); // just to enforce a standard order if two dates are equals
+  }
+}
+
+// TODO this may end up being handled by the enhanced medication request status?
```

Comment:
utility method to display the status we want to display to the user for a prescription or request that is based both on the request status and the dispense status (where the dispense status is the status of the most recent dispense event associated with the order)